test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs

```diff
@@ -160,6 +160,22 @@ public class HubConnectionTests
             Assert.Equal(1, mockProtocol.ParseCalls);
         }
 
+        [Fact]
+        public async Task CheckSyncContext()
+        {
+            SynchronizationContext.SetSynchronizationContext(new SynchronizationContext());
+            var connection = new TestConnection();
+            var hubConnection = new HubConnection(connection, new LoggerFactory());
+            Assert.NotNull(SynchronizationContext.Current);
+            await hubConnection.StartAsync();
+            Assert.Null(SynchronizationContext.Current);
```

Comment:
This looks wrong. That doesn't happen...

Reply:
Yeah, tbh I was kind of confused by this behavior

Reply:
How should we be testing this at the HubConnection level?